File: src/main/java/com/example/the_machine/repo/MessageRepository.java
package com.example.the_machine.repo;

import com.example.the_machine.db.MessageEntity;
import java.util.List;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface MessageRepository extends JpaRepository<MessageEntity, UUID> {

  List<MessageEntity> findByThreadIdOrderByCreatedAt(UUID threadId);

  List<MessageEntity> findByThreadIdOrderByCreatedAtDesc(UUID threadId);
}

File: src/main/kotlin/com/example/the_machine/repo/MessageRepository.kt
package com.example.the_machine.repo

import com.example.the_machine.domain.MessageEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface MessageRepository : JpaRepository<MessageEntity, UUID> {

  fun findByThreadIdOrderByCreatedAt(threadId: UUID): List<MessageEntity>
}